include/git2/patch.h

```diff
@@ -267,6 +267,35 @@ GIT_EXTERN(int) git_patch_to_buf(
 	git_buf *out,
 	git_patch *patch);
 
+/**
+ * Apply a patch to a buffer.
+ *
+ * Attempts to apply the changes represented by a patch to a source buffer
+ * producing a new output buffer. A hunk callback can be given to choose
+ * which hunks are applied.
+ *
+ * @param out The output buffer to be filled
+ * @param filename The output name of the new file
+ * @param mode The output mode of the new file
```

Comment:
Do the out-parameters necessarily have to be non-NULL and if so, does it make sense (hint: they currently have to be non-NULL, but I'm not sure if it should be required)?